agent/src/softwarecontaineragent.cpp

```diff
@@ -241,23 +247,52 @@ void SoftwareContainerAgent::setContainerName(ContainerID containerID, const std
     }
 }
 
-void SoftwareContainerAgent::shutdownContainer(ContainerID containerID)
+bool SoftwareContainerAgent::shutdownContainer(ContainerID containerID)
 {
-    shutdownContainer(containerID, m_softwarecontainerWorkspace->m_containerShutdownTimeout);
+    return shutdownContainer(containerID, m_softwarecontainerWorkspace->m_containerShutdownTimeout);
 }
 
-void SoftwareContainerAgent::shutdownContainer(ContainerID containerID, unsigned int timeout)
+bool SoftwareContainerAgent::shutdownContainer(ContainerID containerID, unsigned int timeout)
 {
     profilefunction("shutdownContainerFunction");
     if (m_shutdownContainers) {
         SoftwareContainer *container = nullptr;
         if (checkContainer(containerID, container)) {
-            container->shutdown(timeout);
+            ReturnCode status = container->shutdown(timeout);
             deleteContainer(containerID);
+
+            if (isError(status)) {
+                log_error() << "Could not shut down container";
+            }
+            return isSuccess(status);
         }
     } else {
         log_info() << "Not shutting down container";
     }
+
+    return false;
+}
+
+bool SoftwareContainerAgent::freezeContainer(ContainerID containerID)
+{
+    SoftwareContainer *container = nullptr;
+    if (checkContainer(containerID, container)) {
+        return isSuccess(container->freeze());
+    }
+
+    log_error() << "Can't freeze container " << containerID;
+    return false;
+}
+
+bool SoftwareContainerAgent::thawContainer(ContainerID containerID)
+{
+    SoftwareContainer *container = nullptr;
+    if (checkContainer(containerID, container)) {
+        return isSuccess(container->thaw());
+    }
+
+    log_error() << "Can't thaw container " << containerID;
+    return false;
```

Comment:
It is a bit weird to do all the error handling in the end. Consider negating the above if clause and do the error handling there instead.